src/mcode-pilot/services/outcomes/CLQOutcomesService.js

```diff
@@ -0,0 +1,216 @@
+        let grade = this.findMcodeElements(pathologyProps, "onco.core.CancerHistologicGrade", true);
+        if (stage) {
+            filter.stage = stage.reference.stage;
+        }
+        if (grade) {
+            filter.grade = grade.reference.getGradeAsSimpleNumber();
+        }
+        filter.tmn = {};
+        if (t) {
+            filter.tmn.t = t.value;
+        }
+        if (n) {
+            filter.tmn.n = n.value;
+        }
+        if (m) {
+            filter.tmn.m = m.value;
+        }
+
+        return filter;
+    }
+
+    /* Build the CLQ tumor markers filter sections based off the tumor makrkers found in the similar
+    patient properties
+    */
+    buildTumorMakersFilter(similarPatientProps) {
+        let pathologyProps = similarPatientProps.pathology.options;
+        let filter = [];
+        // loop over options look for tumor markers and add to filter
+        let markers = this.findMcodeElements(pathologyProps, "onco.core.TumorMarkerTest");
+        for (let x in markers) {
+            let option = markers[x];
+            let code = option.reference.receptorTypeCodeableConcept;
+            let value = option.value;
+            filter.push({
+                code: code.code.code,
+                codeSystem: code.codeSystem.uri,
+                displayName: code.displayText.string,
+                value: value
+            });
+        }
+        return filter;
+    }
+
+    /*
+    Initialize a row of treatment data. This sets the fields that would be expected by the
+    Compass UI to inital default values.
+    */
+    initializeTreatmentData(displayName, treatments) {
+        return {
+            id: _.uniqueId('row_'),
+            displayName,
+            treatments,
+            totalPatients: 0,
+            oneYrSurvival: 0,
+            threeYrSurvival: 0,
+            fiveYrSurvival: 0,
+            sideEffects: {
+                totalReporting: 0,
+                effects: {}
+            }
+        };
+    }
+
+    /* Generate a display name for the Compass Data row based off of a set of treatments*/
+    generateTreatmentDisplayName(treatments) {
+        return treatments.map((treatment) => {
+            return treatment.displayName;
+        }).join(" & ");
+    }
+
+    /* Map an individual CLQ service resonse for a set of treatments into the internal format
+    expected by the Compass UI components. */
+    generateOutcomeData(data) {
+        return data.map((item) => {
+            if (item.total === 0 || !item.sufficiency) {
+                return null;
+            }
+            let row = this.initializeTreatmentData(this.generateTreatmentDisplayName(item.treatments), item.treatments);
+            row.totalPatients = item.total;
+            item.outcomes.forEach((outcome) => {
+                /* eslint-disable */
+                if (outcome.survivalRate == '12') {
+                    row.oneYrSurvival = outcome.total;
```

Comment:
Just so we don't lose track, this will need to be in alignment with the changes made for #679.

Reply:
That is correct, not certain if #679 the way it is implemented will be the final way that task is incorporated.  But yes, will need to be updated once something is added